input-profile.ts

```diff
@@ -0,0 +1,369 @@
+            this._controllerModel.setPositionLocal([0, 0, 0]);
+            this._setComponentsActive(false);
+
+            if (this.addVrModeSwitch)
+                this._controllerModel.addComponent(VrModeActiveSwitch);
+            this.onModelLoaded.notify();
+        }
+        this._cacheGamepadObjectsFromProfile(this._profileJSON, this._controllerModel);
+        this._setHandTrackingControllers(this.defaultController);
+        this.update = () => this._mapGamepadInput();
+    }
+
+    /**
+     * Caches gamepad objects (buttons, axes) from the loaded input profile.
+     * @hidden
+     */
+    private _cacheGamepadObjectsFromProfile(profile: any, obj: Object3D) {
+        const components = profile.layouts[this._handedness].components;
+        if (!components) return;
+
+        this._buttons = [];
+        this._axes = [];
+
+        for (const i in components) {
+            const visualResponses = components[i].visualResponses;
+
+            for (const j in visualResponses) {
+                // update buttons with new interface of current visual response
+                const visualResponse = visualResponses[j];
+                const valueNode = visualResponse.valueNodeName;
+                const minNode = visualResponse.minNodeName;
+                const maxNode = visualResponse.maxNodeName;
+
+                this._gamepadObjects[valueNode] = this._getObjectByName(obj, valueNode);
+                this._gamepadObjects[minNode] = this._getObjectByName(obj, minNode);
+                this._gamepadObjects[maxNode] = this._getObjectByName(obj, maxNode);
+
+                let indice = visualResponses[j].componentProperty;
+                const response: VisualResponse = {
+                    target: this._gamepadObjects[valueNode],
+                    min: this._gamepadObjects[minNode],
+                    max: this._gamepadObjects[maxNode],
+                    id: components[i].gamepadIndices[indice], // Assign a unique ID
+                };
+                switch (indice) {
+                    case 'button':
+                        this._buttons.push(response);
+                        break;
+                    case 'xAxis':
+                        this._axes.push(response);
+                        break;
+                    case 'yAxis':
+                        this._axes.push(response);
+                        break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieves a game object from the specified object based on its name.
+     * @param {Object3D} obj - The object to search for the specified name.
+     * @param {string} name - The name of the object to retrieve.
+     * @return {Object3D|undefined} The retrieved game object, or undefined if not found.
+     * @hidden
+     */
+    private _getObjectByName(obj: Object3D, name: string) {
+        if (!obj || !name) return;
+        const found = obj.findByNameRecursive(name);
+        if (found[0]) return found[0];
```

Comment:
Else?

Not sure if this function adds much value around `findByNameRecursive`. Maybe you just want to inline it where it's used.

Reply:
thats new , first time using  nullish coalescing operator (??) :sweat_smile:  :handshake: 
made it inline :thumbsup: